src/storage/log/aof/log.rs

```diff
@@ -378,4 +378,100 @@ mod tests {
         // Test closing segment
         assert!(a.close().is_ok());
     }
+
+    #[test]
+    fn append_and_read_two_blocks() {
```

Comment:
My quick thought was that `append_read_append` is enough for verifying this issue.
Would you mind clarifying why we need both tests? Is reading two blocks an important distinction?

Reply:
So I am writing in block size of 32kb to disk. There was an issue where if one full block is written, and other one is partially written to, read shouldn't fail spanning two blocks (one on disk and other in memory)

Reply:
[This](https://github.com/surrealdb/surrealkv/blob/0725edbedcf5ac55918abd93c26a05d53f309181/src/storage/log/mod.rs#L1162) should provide more context